components/core/src/config/local/localfilehostconfigprovider.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PluggableFactory} from '../../integrations/pluggablefactory';
+import {TenantNodeConfigProvider} from '../tenantnodeconfigprovider';
+import {StatusError} from '../../util/statuserror';
+import {JsonMap} from '../../util/jsonvalue';
+import path from 'path';
+import fs from 'fs/promises';
+import {TenantNodeConfigJson} from '../tenantnodeconfig';
+import {PluggableFactorySet} from '../../integrations/pluggablefactoryset';
+
+export interface LocalFileMultitenantOptionsJson extends JsonMap {
+  // Path to the base directory of configuration files.
+  basePath: string;
+}
+
+export const LocalFileMultitenantIntegration = {
+  async construct<Allowed extends PluggableFactorySet>(
+    json: LocalFileMultitenantOptionsJson
+  ): Promise<LocalFileHostConfigProvider<Allowed>> {
+    return new LocalFileHostConfigProvider(json);
+  },
+};
+
+export class LocalFileHostConfigProvider<Allowed extends PluggableFactorySet>
+  implements TenantNodeConfigProvider<Allowed>
+{
+  readonly type = 'tenantConfigProvider';
+  private basePath: string;
+
+  constructor(options: LocalFileMultitenantOptionsJson) {
+    this.basePath = options.basePath;
+  }
+
+  async getTenantConfig(
+    hostId: string
+  ): Promise<TenantNodeConfigJson<Allowed>> {
+    const configFilePath = path.resolve(
+      './' + hostId + '/config.json',
```

Comment:
./ isn't a problem. It will be resolved relative to basePath. If basePath were /mnt/configs, then the ./ would be resolved to a slash. Dot slash also allows us to reference the current directory by passing in an empty basePath, so it's very slightly more flexible.